pkgs/applications/version-management/bcompare/default.nix

```diff
@@ -0,0 +1,67 @@
+{ autoPatchelfHook, bzip2, cairo, coreutils, fetchurl, gdk_pixbuf, gnome2, gtk2, kcoreaddons, ki18n, kio, kservice, lib, qt4, qt511, qtbase, stdenv }:
+
+stdenv.mkDerivation rec {
+  pname = "bcompare";
+  version = "4.2.10.23938";
+
+  src = fetchurl {
+    url = "https://www.scootersoftware.com/${pname}-${version}_amd64.deb";
+    sha256 = "1825s1lbaj20d712czmaaqg7mkwfb650r53af2y30j29p6yd4wal";
+  };
+
+  unpackPhase = ''
+    ar x $src
+    tar xfz data.tar.gz
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin $out/lib $out/share
+    cp -R usr/share $out/
+    cp -R usr/lib $out/
+    cp -R usr/bin $out/
+
+    # Remove library that refuses to be autoPatchelf'ed
+    rm $out/lib/beyondcompare/ext/bcompare_ext_kde.amd64.so
+
+    substituteInPlace $out/bin/bcompare \
+      --replace "/usr/lib/beyondcompare" "$out/lib/beyondcompare" \
+      --replace "/bin/bash" "/usr/bin/env bash"
```

Comment:
This part here will be done by the patchshebangs hook, and it would also make it an absolute path to bash in the nix store.

Reply:
This does not fix shebang. It's a call of bash in the script on line 43.

Reply:
In this case we shouldn't use `/usr/bin/env bash` because that's impure.

Please substitute with `runtimeShell`.

Reply:
I would like to have 'bash' there. Will 'runtimeShell' be always 'bash'?

Reply:
It's bash.

See the definition of `runtimeShell` in `all-packages.nix`.

Reply:
Thank you! I'll ripgrep nixpkgs more often.

Reply:
done, thx!!